Marlin/src/gcode/geometry/G92.cpp

```diff
@@ -111,29 +138,60 @@ void GcodeSuite::G92() {
       }
 
       current_position[Z_AXIS] = curr_z;
-      if(activePrimaryZTO)
-        primaryZTO += adjust_z;
+
+      if(active_extruder == 0)
+          primaryZTO += adjust_z;
       else
         secondaryZTO += adjust_z;
+
     }
 
     //report
     SERIAL_ECHO_START();
     SERIAL_ECHOLNPAIR("adjust_z(now): ", adjust_z);
     SERIAL_ECHOLNPAIR("primaryZTO: ", primaryZTO);
     SERIAL_ECHOLNPAIR("secondaryZTO: ", secondaryZTO);
+    SERIAL_ECHOLNPAIR("PrimaryZTOEprom: ", primaryZTOEprom);
+    SERIAL_ECHOLNPAIR("SecondaryZTOEprom: ", secondaryZTOEprom);
+  }
+
+  if(parser.seen('I')){
+    float curr_z = current_position[Z_AXIS];
+    if((active_extruder == 0) && (curr_z >= -primaryZTO) ){
+      adjust_z = primaryZTOEprom - primaryZTO;
+      if(adjust_z == 0){
+        //nothing
+      } else {
+        doZAdjust = true;
+        if(adjust_z > 0){
+          do_blocking_move_to_z(curr_z + Z_ADJUST_HOP_DISTANCE + adjust_z, HOMING_FEEDRATE_Z);
+          do_blocking_move_to_z(curr_z + adjust_z, HOMING_FEEDRATE_Z);
+
+        } else {//adjust_z < 0
+          do_blocking_move_to_z(curr_z + Z_ADJUST_HOP_DISTANCE, HOMING_FEEDRATE_Z);
+          do_blocking_move_to_z(curr_z + adjust_z, HOMING_FEEDRATE_Z);
+        }
+
```

Comment:
Perhaps cleaner and more readable as:
```c++
float hop_position = std::max(0, adjust_z) + curr_z + Z_ADJUST_HOP_DISTANCE;

do_blocking_move_to_z(hop_position, HOMING_FEEDRATE_Z);
do_blocking_move_to_z(curr_z + adjust_z, HOMING_FEEDRATE_Z);
```

Reducing the number of pathways code can take is almost always a preferable choice.

Reply:
only one i didn't address in my cleanup comments.